java/spark/v3.5/src/test/java/com/lancedb/lance/spark/SparkCatalogTest.java

```diff
@@ -0,0 +1,122 @@
+ * limitations under the License.
+ */
+
+package com.lancedb.lance.spark;
+
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.RowFactory;
+import org.apache.spark.sql.catalyst.analysis.NoSuchTableException;
+import org.apache.spark.sql.catalyst.analysis.TableAlreadyExistsException;
+import org.apache.spark.sql.types.StructType;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.DataTypes;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import java.nio.file.Path;
+
+public class SparkCatalogTest {
+  @TempDir
+  static Path tempDir;
+
+  private static SparkSession spark;
+
+  @BeforeAll
+  static void setup() {
+    spark = SparkSession.builder()
+        .appName("SparkCatalogTest")
+        .master("local")
+        .config("spark.sql.catalog.dev", "com.lancedb.lance.spark.SparkCatalog")
+        .config("spark.sql.catalog.dev.warehouse", tempDir.toString())
+        .getOrCreate();
+  }
+
+  @AfterAll
+  static void tearDown() {
+    if (spark != null) {
+      spark.stop();
+    }
+  }
+
+  @Test
+  public void testSparkCreate() {
+    String tableName = "dev.db.lance_create_table";
+    // Workflow: loadTable with NoSuchTable -> createTable -> loadTable
+    spark.sql("CREATE TABLE " + tableName + " (id INT, data STRING) USING lance");
+    // Workflow: [Gap] dropTable
+    // spark.sql("DROP TABLE " + tableName);
+  }
+
+  @Test
+  @Disabled
+  public void testSparkInsert() {
+    String tableName = "dev.db.lance_insert_table";
+    spark.sql("CREATE TABLE " + tableName + " (id INT, data STRING) USING lance");
+    // Workflow: loadTable -> SparkTable -> [Gap]SparkTable.getSchema
+    // -> [Gap] SparkTable.partitioning() returns no partition. empty Transform[]
+    // -> [Gap] SparkTable.capabilities() returns capabilities e.g. batch read/write/streaming write
+    // -> [Gap] SparkTable.newWriteBuilder -> BatchWrite -> lance append
+    spark.sql("INSERT INTO " + tableName + " VALUES (1, 'a'), (2, 'b')");
+    // Workflow: loadTable -> SparkTable -> [Gap]SparkTable.getSchema
+    // -> [Gap] SparkTable.capabilities() -> [Gap]SparkTable.newScanBuilder()
+    // -> [Gap] SparkScanBuilder.pushAggregation().build()
+    // -> [Gap] LocalScan.readSchema() -> [Gap] LocalScan.rows[]
+    spark.sql("SELECT * FROM " + tableName).show();
+    spark.sql("SELECT COUNT(*) FROM " + tableName).show();
+  }
+
+  @Test
+  @Disabled
+  public void testSparkDataFrameCreate() throws TableAlreadyExistsException {
+    Dataset<Row> data = createSparkDataFrame();
+    String tableName = "dev.db.lance_df_table";
+    // Same as create + insert
+    data.writeTo(tableName).using("lance").create();
+    spark.table(tableName).show();
```

Comment:
Could you also check that manifest files have been created under the directory?

Reply:
Will add to the test! thanks